agent/doc/UML/agent_device_management_design.puml

```diff
@@ -1,32 +1,103 @@
+class DeviceStatusListenerManager{
+    List<DeviceStatusListener> deviceStatusListeners
+    registerDeviceStatusListener(DeviceStatusListener)
+    onDeviceInactive(DeviceInfo)
+    onDeviceActive(DeviceInfo)
+}
+class TestTaskEngineService{
+    TestRunDevice chooseDevice(TestTask)
+}
+
+
+class DeviceInfo
+
+class TestRunDeviceOrchestrator{
+    void setup(TestRunDevice, Logger);
+    void unset(TestRunDevice, Logger);
+    void killAll(TestRunDevice);
+    void getAppiumDriver(TestRunDevice, Logger);
+    void quitAppiumDriver(TestRunDevice, Logger);
+    void wakeUpDevice(TestRunDevice, Logger);
+    boolean installApp(TestRunDevice,String packagePath, Logger);
+    boolean uninstallApp(TestRunDevice, String packageName, Logger);
+    void resetPackage(TestRunDevice, String packageName, Logger);
+    void startScreenRecorder(TestRunDevice, File folder, int maxTimeInSecond, Logger);
+    void stopScreenRecorder(TestRunDevice);
+    void startLogCollector(TestRunDevice, String pkgName, @NotNull TestRun testRun, Logger);
+    void stopLogCollector(TestRunDevice);
+    void setRunningTestName(TestRunDevice, String runningTestName);
+    void addCurrentTask(TestRunDevice, TestTask);
+    void finishTask(TestRunDevice);
+    void quitMobileAppiumDriver(TestRunDevice, Logger);
+    void updateScreenshotImageAsyncDelay(TestRunDevice, long delayMillis, FileAvailableCallback, Logger);
+    void grantAllTaskNeededPermissions(TestRunDevice, TestTask, Logger);
+}
+
+
+class TestRunDevice{
+    String Tag
+    DeviceInfo deviceInfo;
+    WebDriver appiumDriver;
+    ScreenRecorder screenRecorder;
+    LogCollector logCollector;
+    Map<Thread, String> currentCommand
+    Map<Thread, String> currentProcess
+    Map<Thread, TestTask> currentTask
+}
+
+class TestRunDeviceCombo extends TestRunDevice{
+    List<TestRunDevice> pairedDevices
+    TestRunDevice getDeviceByTag(String tag)
+    List<TestRunDevice> getDevices()
+}
+Class TestRun{
+    TestRunDevice device
+}
+class TestRunner{
+    runTest(TestTask,TestRunDevice)
+}
+
+interface IDeviceDriver
+class DeviceDriverManager implements IDeviceDriver{
+    Map<DeviceType, IDeviceDriver> deviceDrivers
+}
+abstract class AbstractDeviceDriver implements IDeviceDriver
+class AndroidDeviceDriver extends AbstractDeviceDriver
+class IOSDeviceDriver   extends AbstractDeviceDriver
+class WindowsDeviceDriver extends AbstractDeviceDriver
 
-interface TestDriver
+TestRunner --> TestRun : create
+TestRun ---> TestRunDevice
+TestRunner -r--> TestRunDeviceOrchestrator
 
-AndroidTestDeviceManager --|> TestDeviceManager
-IOSTestDeviceManager --|> TestDeviceManager
-WindowsTestDeviceManager --|> TestDeviceManager
+TestRunDeviceOrchestrator --> DeviceDriverManager
 
-AgentManagementService *--> AndroidTestDeviceManager
-AgentManagementService *--> IOSTestDeviceManager
-AgentManagementService *--> WindowsTestDeviceManager
+TestTaskEngineService --> TestRunner
 
-TestDeviceManager --> TestDriver
+IDeviceDriver -> DeviceInfo
```

Comment:
Could you also list the method of `IDeviceDriver`? Like what we did with `TestRunDeviceOrchestrator`.